### PR TITLE
feat: support Dust tree by age

### PR DESCRIPTION
**Pull request by wugeer**
# Background

 Dust tree by age https://github.com/bootandy/dust/issues/385
Add a new feature to dust to show the tree by age          

# Full Changelogs

- feat: support Dust tree by age

# Issue Reference

fix:   Dust tree by age https://github.com/bootandy/dust/issues/385

# Test Result
dust help command output
image
cargo test run test results

image
image
image
image

show by accesstime
image
show by modifiedtime
image
show by createdtime
image

### CODE REVIEW

File: src/display.rs [hunk 1]
```diff
@@ -389,6 +399,17 @@ fn get_pretty_size(node: &DisplayNode, is_biggest: bool, display_data: &DisplayD
...
+fn get_pretty_file_modified_time(timestamp: i64) -> String {
+    // println!("{}", timestamp);
+    let datetime: DateTime<Utc> = Utc.timestamp_opt(timestamp, 0).unwrap();
+
+    // 根据本地时区进行转换
+    let local_datetime = datetime.with_timezone(&Local);
+
+    // 按照指定格式格式化为字符串
+    local_datetime.format("%Y%m%dT%H:%M:%S").to_string()
+}
```
Comment (bootandy) [inline src/display.rs:411]: Maybe: "%Y-%m-%dT%H:%M:%S"

Also lets get rid of the comments.

**Review by bootandy — COMMENTED** [review]
A quick scan says it looks sensible. I am a bit worried about performance though 

I created: https://github.com/bootandy/dust/issues/415 to track down the performance issues with a similar PR.

**Comment by wugeer** [comment]
> A quick scan says it looks sensible. I am a bit worried about performance though
> 
> I created: #415 to track down the performance issues with a similar PR.

I will test my submitted performance test, if the test is OK, I will @you, please review the code.

File: src/display.rs (hunk 1 above)
Comment (wugeer) [reply src/display.rs:411]: I will resubmit the code after testing the performance and fix this together

Comment (wugeer) [reply src/display.rs:411]: Yeah, you are right! I will conduct performance testing on my current code submission, modify the time format, and remove the unnecessary comments before resubmitting the code.

**Comment by wugeer** [comment]
> A quick scan says it looks sensible. I am a bit worried about performance though
> 
> I created: #415 to track down the performance issues with a similar PR.

Hello, Tests with hyperfine show that this PR means we are running ~ 4% slower than the v1.0.0 release. I'm not sure if the performance test results are okay. Here are my test results.
image
image
image
image
I'm not sure if my performance tests are sufficient. If more information is needed, please let me know, and I will provide it as soon as possible. :) @bootandy

**Comment by bootandy** [comment]
I think this is sensible, I just want to sleep on it for a bit.

**Comment by bootandy** [comment]
>Hello, Tests with hyperfine show that this PR means we are running ~ 4% slower than the v1.0.0 release. I'm not sure if the performance test results are okay. Here are my test results.

My hyperfine tests show this branch is about 4% faster ! 

Either way its fine, I'm looking for drops of >10% before I get nervous.

**Comment by bootandy** [comment]
Why did you choose the flag `-m` ? It's fine but I'd like to copy an existing tool that uses a similar flag if possible

@wugeer

**Comment by wugeer** [comment]
> Why did you choose the flag `-m` ? It's fine but I'd like to copy an existing tool that uses a similar flag if possible This is fine, but if possible I'd like to copy an existing tool that uses a similar flag
>
> @wugeer

hello， I chose randomly from the remaining single-letter parameters. I originally wanted to use -t but it was already taken. Do you have any other good suggestions? @bootandy

**Comment by bootandy** [comment]
I don't have other suggestions, I'll research some other tools to see if anyone has anything similar, but we'll go with -m if I can't find anything.

**Pull request merged**